src/emulation/mod.rs

```diff
@@ -62,28 +62,34 @@ impl SyscallMock {
     }
 
     fn handle_step(&mut self, received: protocol::Command) -> R<PathBuf> {
-        let stdout = match self.expected.steps.pop_front() {
-            Some(next_expected_step) => {
-                if next_expected_step.command != received {
-                    self.register_error(&next_expected_step.command.format(), &received.format());
+        let mock_config = match self.protocol.steps.pop_front() {
+            Some(next_protocol_step) => {
+                if next_protocol_step.command != received {
+                    self.register_error(&next_protocol_step.command.format(), &received.format());
+                }
+                executable_mock::Config {
+                    stdout: next_protocol_step.stdout,
+                    exitcode: next_protocol_step.exitcode,
                 }
-                next_expected_step.stdout
             }
             None => {
                 self.register_error("<protocol end>", &received.format());
-                vec![]
+                executable_mock::Config {
```

Comment:
No, it's not just a dummy struct. Currently the script is not stopped if any errors are detected.

So in this branch the protocol has ended, but the script is attempting to execute more subprocesses. An error about that is registered. Then the syscall `exec`ing the subprocess is still executed (outputting nothing to `stdout` and returning with a `0` exitcode). After that, the script under test is not stopped. It might even execute further subprocesses, that will all go through this branch. (The `register_error` function only ever remembers the first error.)

This is done like this, because I didn't want to figure out how to shut down the tested script and so far I didn't see examples for where this behaviour is bad. Any idea on how to make it clearer in the code that this is how it works? (That'd be in a different PR, I guess.)

Reply:
got it, maybe just giving this a name would make it clearer? something like `executable_mock::allow_script_to_procede()`. alternatively, the error message above could be more informative: `protocol ended but script under test continued, returning exit 0 for all remaining calls`.

Reply:
I don't really want to change the error messages to the user. Ideally it should be irrelevant to the user that the script is continued.

Which function could be called `allow_script_to_procede`? A function that returns `executable_mock::Config { stdout: vec![], exitcode: 0 }`? I think that'd be weird, since that's just kind of the default `Config`. But maybe I'm misunderstanding?